### PR TITLE
formatted gem completion

### PR DESCRIPTION
**Pull request by NoahGorny**
Cleanup gem completion as part of #1782

### CODE REVIEW

File: completion/available/gem.completion.bash [hunk 1]
```diff
@@ -1,41 +1,34 @@
-#!/usr/bin/env bash
+# shellcheck shell=bash
 # Completion for gem
 
-_installcomp() {
-  if [ -z "$REMOTE_GEMS" ]
-  then
-    REMOTE_GEMS=( $(gem list --remote --no-versions | tr '\n' ' ') )
-  fi
+__gem_completions() {
```
Comment (cornfeedhobo) [inline completion/available/gem.completion.bash:4]: could we drop the plural to `__gem_completion`?

File: completion/available/gem.completion.bash [hunk 2]
```diff
@@ -1,41 +1,34 @@
...
+			return 0
+			;;
+	esac
+	local commands=(build cert check cleanup contents dependency environment fetch generate_index help install list lock outdated owner pristine push query rdoc search server sources specification stale uninstall unpack update which)
+	# shellcheck disable=SC2207
+	COMPREPLY=($(compgen -W "${commands[*]}" -- "$cur"))
 }
 
-complete -F _gem gem
+complete -F __gem_completions gem
```
Comment (cornfeedhobo) [inline completion/available/gem.completion.bash:34]: same as above.

File: completion/available/gem.completion.bash (hunk 1 above)
Comment (NoahGorny) [reply completion/available/gem.completion.bash]: fixed, nice idea!

File: completion/available/gem.completion.bash (hunk 2 above)
Comment (NoahGorny) [reply completion/available/gem.completion.bash]: fixed

**Review by cornfeedhobo — APPROVED** [review]

**Pull request merged**